javalib/src/main/scala/java/util/Formatter.scala

```diff
@@ -79,1587 +91,989 @@ class Formatter(
-            formatToken.getStrFlags(),
-            formatToken.getConversionType())
-        else if ('o' == currentConversionType) {
-          result.append("0")
-          startIndex += 1
-        } else {
-          result.append("0x")
-          startIndex += 2
-        }
-      }
-
-      if (formatToken.isFlagSet(FormatToken.FLAG_MINUS) && formatToken
-            .isFlagSet(FormatToken.FLAG_ZERO))
-        throw new IllegalFormatFlagsException(formatToken.getStrFlags())
-
-      if (value < 0)
-        isNegative = true
-
-      if ('d' == currentConversionType) {
-        val numberFormat = getNumberFormat()
-        if (formatToken.isFlagSet(FormatToken.FLAG_COMMA))
-          numberFormat.setGroupingUsed(true)
-        else
-          numberFormat.setGroupingUsed(false)
-        result.append(numberFormat.format(arg))
-      } else {
-        val BYTE_MASK: Long  = 0x00000000000000FFL;
-        val SHORT_MASK: Long = 0x000000000000FFFFL;
-        val INT_MASK: Long   = 0x00000000FFFFFFFFL;
-        if (isNegative) {
-          if (arg.isInstanceOf[Byte])
-            value &= BYTE_MASK
-          else if (arg.isInstanceOf[Short])
-            value &= SHORT_MASK
-          else if (arg.isInstanceOf[Int])
-            value &= INT_MASK
-        }
-        if ('o' == currentConversionType)
-          result.append(java.lang.Long.toOctalString(value))
-        else
-          result.append(java.lang.Long.toHexString(value))
-        isNegative = false
+          formatNullOrThrowIllegalFormatConversion()
       }
-
-      if (!isNegative) {
-        if (formatToken.isFlagSet(FormatToken.FLAG_ADD)) {
-          result.insert(0, '+')
-          startIndex += 1
-        }
-        if (formatToken.isFlagSet(FormatToken.FLAG_SPACE)) {
-          result.insert(0, ' ')
-          startIndex += 1
-        }
-      }
-
-      if (isNegative && formatToken.isFlagSet(FormatToken.FLAG_PARENTHESIS)) {
-        result = wrapParentheses(result)
-        return result.toString()
-      }
-      if (isNegative && formatToken.isFlagSet(FormatToken.FLAG_ZERO))
-        startIndex += 1
-      return padding(result, startIndex)
     }
 
-    private def wrapParentheses(result: StringBuilder): StringBuilder = {
-      // delete the '-'
-      result.deleteCharAt(0)
-      result.insert(0, '(')
-      if (formatToken.isFlagSet(FormatToken.FLAG_ZERO)) {
-        formatToken.setWidth(formatToken.getWidth() - 1)
-        padding(result, 1)
-        result.append(')')
-      } else {
-        result.append(')')
-        padding(result, 0)
-      }
-      result
-    }
+    (conversion: @switch) match {
+      case 'b' | 'B' =>
+        validateFlags(flags,
+                      conversion,
+                      invalidFlags = NumericOnlyFlags | AltFormat)
+        val str =
+          if ((arg.asInstanceOf[AnyRef] eq false
+                .asInstanceOf[AnyRef]) || arg == null) "false"
+          else "true"
```

Comment:
This way of testing things is too specific to Scala.js. We should do the following:
```scala
val str = arg match {
  case arg: java.lang.Boolean => arg.toString()
  case null                   => "false"
  case _                      => "true"
}
```